Bump cardano-node to 1.35.3-rc1

sevanspowell opened the pull request:
- Bump dependencies:
  - cardano-node to 1.35.3-rc1
  - hw-aeson to match cardano-node
  - hedgehog-extras to match cardano-node
- Update compatibility matrix.

### Comments

Changelogs:

- [API](https://github.com/input-output-hk/cardano-node/blob/release/1.35/cardano-api/ChangeLog.md)
- [CLI](https://github.com/input-output-hk/cardano-node/blob/release/1.35/cardano-cli/ChangeLog.md)
- [Node](https://github.com/input-output-hk/cardano-node/blob/release/1.35/cardano-node/ChangeLog.md)

- [Compare v1.35.3-testnetonly and v1.35.3-rc1](https://github.com/input-output-hk/cardano-node/compare/1.35.3-testnetonly...1.35.3-rc1)

### Issue Number

ADP-2123

sevanspowell committed:
Bump cardano-node to 1.35.3-rc1

- Bump dependencies:
  - cardano-node to 1.35.3-rc1
  - hw-aeson to match cardano-node
  - hedgehog-extras to match cardano-node
- Update compatibility matrix.

sevanspowell committed:
Bump cardano-addresses and hw-aeson

- Bump hw-aeson to v0.1.6.0.
- Bump cardano-addresses to fix issue with JM.toList.
  - See https://github.com/input-output-hk/cardano-wallet/pull/3428#issuecomment-1208841366

sevanspowell committed:
Update cardano-wallet to compile with 1.35.3-rc1

- Include new protocol parameter "UTxOCostPerByte".

paweljakubas reviewed (approved):
LGTM, but I would wait with merging cardano-addresses, have the final commit and use it here

jonathanknowles commented on lib/shelley/test/unit/Cardano/Wallet/Shelley/TransactionSpec.hs:
```diff
@@ -3855,6 +3855,7 @@ mockProtocolParametersForBalancing = (mockProtocolParameters, nodePParams)
             Just $ Cardano.ExecutionUnits 10000000000 14000000
         , Cardano.protocolParamCollateralPercent = Just 1
         , Cardano.protocolParamMaxCollateralInputs = Just 3
+        , Cardano.protocolParamUTxOCostPerByte = Just 4310
```
Suggestion: locate the following protocol parameters together:
```hs
        , Cardano.protocolParamUTxOCostPerWord = Just 34482
        , Cardano.protocolParamUTxOCostPerByte = Just 4310
```

jonathanknowles replied:
Suggestion: given that these are commonly-used test values, you could just import the values defined here:

https://github.com/input-output-hk/cardano-wallet/blob/1f72c95d081c973e492da739faeee4bdefcfd244/lib/core/src/Cardano/Wallet/Primitive/Types/MinimumUTxO/Gen.hs#L154
https://github.com/input-output-hk/cardano-wallet/blob/1f72c95d081c973e492da739faeee4bdefcfd244/lib/core/src/Cardano/Wallet/Primitive/Types/MinimumUTxO/Gen.hs#L162

jonathanknowles commented on lib/shelley/src/Cardano/Wallet/Shelley/Network/Blockfrost.hs:
```diff
@@ -965,6 +965,7 @@ fromBlockfrostPP network pp@BF.ProtocolParams{..} = do
                     , protocolParamCollateralPercent = Just collateralPercent
                     , protocolParamMaxCollateralInputs =
                         Just $ intCast maxCollateralInputs
+                    , protocolParamUTxOCostPerByte = Just 4310
```
Suggestion: locate this protocol parameter value next to `protocolParamUTxOCostPerWord`.

jonathanknowles replied:
Suggestion: given that these are commonly-used test values, you could just import the values defined here:

https://github.com/input-output-hk/cardano-wallet/blob/1f72c95d081c973e492da739faeee4bdefcfd244/lib/core/src/Cardano/Wallet/Primitive/Types/MinimumUTxO/Gen.hs#L154
https://github.com/input-output-hk/cardano-wallet/blob/1f72c95d081c973e492da739faeee4bdefcfd244/lib/core/src/Cardano/Wallet/Primitive/Types/MinimumUTxO/Gen.hs#L162

jonathanknowles reviewed (commented):
Hi Sam

Just a couple of suggestions that may help readability/discoverability of minimum-UTxO-related protocol parameter values.

sevanspowell committed:
Co-locate protocolUpdateUTxO parameters

- Co-locate protocolUpdateUTxO parameters.
- Import definitions for these parameters from
  Cardano.Wallet.Primitive.Types.MinimumUTxO.Gen.

sevanspowell committed:
Update Blockfrost fromProtocolParameters

- Blockfrost has not yet updated their API to include
  "protocolParamUTxOCostPerByte". Therefore, when importing protocol parameters
  from Blockfrost, we set this field to Nothing.

jonathanknowles commented on lib/shelley/test/unit/Cardano/Wallet/Shelley/TransactionSpec.hs:
```diff
@@ -3842,7 +3846,10 @@ mockProtocolParametersForBalancing = (mockProtocolParameters, nodePParams)
+        , Cardano.protocolParamUTxOCostPerWord =
+            Just $ fromIntegral $ SL.unCoin testParameter_coinsPerUTxOWord_Alonzo
+        , Cardano.protocolParamUTxOCostPerByte =
+            Just $ fromIntegral $ SL.unCoin testParameter_coinsPerUTxOByte_Babbage
```
👍🏻

jonathanknowles commented on lib/shelley/test/unit/Cardano/Wallet/Shelley/TransactionSpec.hs:
```diff
@@ -3842,7 +3846,10 @@ mockProtocolParametersForBalancing = (mockProtocolParameters, nodePParams)
         , Cardano.protocolParamTreasuryCut  = 0
-        , Cardano.protocolParamUTxOCostPerWord = Just 34482
+        , Cardano.protocolParamUTxOCostPerWord =
+            Just $ fromIntegral $ SL.unCoin testParameter_coinsPerUTxOWord_Alonzo
```
It might be possible to use `intCast` here, which checks statically that the conversion is lossless. (Instead of `fromIntegral`.)

sevanspowell committed:
Upgrade hw-aeson

sevanspowell committed:
Use sevanspowell fork until hw-aeson is merged
